mne/channels/montage.py

```diff
@@ -649,10 +649,19 @@ def read_dig_montage(hsp=None, hpi=None, elp=None, point_names=None,
                         'Right periauricular point': 'rpa',
                         'Left periauricular point': 'lpa'}
 
+        scale = dict(mm=1e-3, cm=1e-2, auto=1e-2, m=None)
+        if unit not in scale:
+            raise ValueError("Unit needs to be one of %s, not %r" %
+                             (tuple(map(repr, scale)), unit))
+
         for s in sensors:
             name, number, kind = s[0].text, int(s[1].text), int(s[2].text)
             coordinates = np.array([float(s[3].text), float(s[4].text),
                                     float(s[5].text)])
+
+            if scale[unit] is not None:
```

Comment:
I made the changes you suggested for my additions since I agree they are much cleaner. I didn't change the other code since I wanted to keep the changes restricted to the bug fix. Let me know if you would prefer otherwise.